src/nprintml/net/step.py

```diff
@@ -0,0 +1,311 @@
+            '-p', '--payload',
+            metavar='INTEGER',
+            type=int,
+            help="include n bytes of payload",
+        )
+        self.group_parser.add_argument(
+            '-P', '--pcap-file', '--pcap_file',
+            default=(),
+            metavar='FILE',
+            nargs='*',
+            type=FileAccessType(os.R_OK),
+            help="pcap infile",
+        )
+        self.group_parser.add_argument(
+            '--pcap-dir', '--pcap_dir',
+            default=(),
+            metavar='DIR',
+            nargs='*',
+            type=DirectoryAccessType(ext='.pcap'),
+            help="directory containing pcap infile(s) with file extension '.pcap'",
+        )
+        self.group_parser.add_argument(
+            '-R', '--relative-timestamps', '--relative_timestamps',
+            action='store_true',
+            help="include relative timestamp field",
+        )
+        self.group_parser.add_argument(
+            '-t', '--tcp',
+            action='store_true',
+            help="include tcp headers",
+        )
+        self.group_parser.add_argument(
+            '-u', '--udp',
+            action='store_true',
+            help="include udp headers",
+        )
+
+    @staticmethod
+    def get_output_directory(args):
+        return args.outdir / 'nprint'
+
+    @classmethod
+    def make_output_directory(cls, args):
+        outdir = cls.get_output_directory(args)
+        outdir.mkdir()
+        return outdir
+
+    def generate_argv(self, args, pcap_file=None):
+        """Construct arguments for `nprint` command."""
+        # generate shared/global arguments
+        if args.verbose:
+            yield '--verbose'
+
+        # support arbitrary pcap infile(s)
+        if pcap_file:
+            yield from ('--pcap_file', pcap_file)
+
+        # add group (nPrint-specific) arguments
+        for action in self.group_parser._group_actions:
+            if action.dest in ('pcap_file', 'pcap_dir'):
+                continue
+
+            key = action.option_strings[-1]
+            value = getattr(args, action.dest)
+
+            if value is not action.default:
+                yield key
+
+                if not isinstance(value, bool):
+                    yield value
+
+        # add output path
+        outdir = self.get_output_directory(args)
+        outname_stem = pathlib.Path(pcap_file).stem if pcap_file else 'netcap'
+        outpath = outdir / f'{outname_stem}.npt'
+        if outpath.exists():
+            last_paths = sorted(outdir.glob(f'{outname_stem}.[0-9][0-9][0-9].npt'), reverse=True)
+            if last_paths:
+                last_match = re.fullmatch(outname_stem + r'\.(\d{3})\.npt', last_paths[0])
+                last_subext = last_match.group(1)
+                subext = int(last_subext) + 1
+            else:
+                subext = 1
+
+            outpath = outdir / f'{outname_stem}.{subext:03d}.npt'
+            assert not outpath.exists()
```

Comment:
Sort of a similar discussion to the last one about generating output paths. Here I just thought I'd try to make nprint file names reflect the pcap file names from which they were generated. Obviously this is likely to be less important, but a nicety, at least for debugging.